### PR TITLE
Domain Manager role standard (issues#184)

### PR DESCRIPTION
**Pull request by markus-hentsch**
Adds a standard for the Keystone API policy configuration to introduce the `domain-manager` role as per  SovereignCloudStack/issues#184

It includes extensions for Keystone groups and therefore closes SovereignCloudStack/issues#383

### CODE REVIEW

**Comment by fkr** [comment]
Reviews more than welcome.

File: Standards/scs-0302-v1-domain-manager-role.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+track: IAM
+---
+
+## Introduction
+
+SCS Clouds should provide a way to grant Domain Manager rights to SCS Customers which provides IAM self-service capabilities within an OpenStack domain.
+This is not properly implemented in the default OpenStack configuration and requires specific adjustments to the Keystone identity management configuration.
+To avoid conflict with the unscoped `admin` role in OpenStack we want to refer to this new role as "Domain Manager" (`domain-manager`).
+
+### Glossary
```
Comment (anjastrunk) [inline Standards/scs-0302-v1-domain-manager-role.md:14]: I like your glossary. :-) We should add `Glossary` as additional mandatory section in our [standard for standards](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0001-v1-sovereign-cloud-standards.md).

**Review by anjastrunk — APPROVED** [review]
A standard documented excellently.

**Review by artificial-intelligence — APPROVED** [review]
LGTM; I guess you can remove the "WIP".

**Comment by markus-hentsch** [comment]
@anjastrunk, @artificial-intelligence thank you for reviewing this draft!

> I guess you can remove the "WIP"

Not yet. I'm still busy implementing extensive tests (on a dedicated [domain-manager-role-tests](https://github.com/SovereignCloudStack/standards/tree/domain-manager-role-tests) branch) that are able to validate the conformance of the standard and secure implementation of the policy via the API. While doing so I'm still refining the policy definitions due to edge cases I'm discovering during test implementation.

**Review by josephineSei — COMMENTED** [review]
Overall this looks good to me. Inline I have a few comments

**Comment by markus-hentsch** [comment]
I'm unable to fix the last two linter errors because the links that it complains being dead will only become valid once the PR is merged.

**Review by josephineSei — APPROVED** [review]
